### PR TITLE
Add game form

### PR DESCRIPTION
**Pull request by MattyGroch**
This is a bunch of UI improvements, along with the addition of the Add Game form, which works splendidly (aside from a minor issue in duplicate validation).

I think there may be some minor issues with the models.py because my OCD behaviors took care of some sanitizing. Sorry.

### CODE REVIEW

File: gamehype/forms.py [hunk 1]
```diff
@@ -28,3 +36,21 @@ def validate_email(self, email):
...
+        [validators.required(), validators.length(min=2, max=128)]
+        )
+    release_date = DateField('Release Date', format='%Y-%m-%d')
+    genre = SelectField(
+        'Genre(s)',
+        choices=[('Cool', 'Cool'), ('Uncool', 'Uncool')]
+        )
+    submit = SubmitField('Submit')
+
+    def validate_game_title(self, game_name):
```
Comment (Poddo) [inline gamehype/forms.py]:
```suggestion
    def validate_game_name(self, game_name):
```

Comment (Poddo) [reply gamehype/forms.py]: validate_  needs to match the name of the form field

Comment (Poddo) [reply gamehype/forms.py]: (I did this locally and it works)

File: gamehype/forms.py [hunk 2]
```diff
@@ -28,3 +36,21 @@ def validate_email(self, email):
         user = User.query.filter_by(email=email.data).first()
         if user is not None:
             raise ValidationError('Please use a different email address.')
+
+
+class AddGameForm(FlaskForm):
+    game_name = StringField(
+        'Game Title',
+        [validators.required(), validators.length(min=2, max=128)]
```
Comment (Poddo) [inline gamehype/forms.py]: this is weird? I'm not sure if its a byproduct of you trying to make the validator fire, but the format here is much different than every other field. They don't have `validators.required` etc`

Comment (Poddo) [inline gamehype/forms.py]:
```suggestion
        validators=[DataRequired(), validators.length(min=2, max=128)]
```

**Review by Poddo — CHANGES_REQUESTED** [review]
Check this out, it fixes it locally for me

File: gamehype/forms.py (hunk 1 above)
Comment (MattyGroch) [reply gamehype/forms.py]: Ah, that makes sense.

File: gamehype/forms.py (hunk 2 above)
Comment (MattyGroch) [reply gamehype/forms.py]: I was just going off of another guide that had multiple validators listed out this way. If this still works locally, I'll keep it for consistency.

**Review by Poddo — APPROVED** [review]
works for me

**Pull request merged**